core/amber/src/main/scala/edu/uci/ics/amber/engine/architecture/controller/WorkflowScheduler.scala

```diff
@@ -0,0 +1,66 @@
+package edu.uci.ics.amber.engine.architecture.controller
+
+import edu.uci.ics.amber.engine.architecture.scheduling.{
+  ExpansionGreedyRegionPlanGenerator,
+  Region,
+  RegionIdentity,
+  RegionPlan
+}
+import edu.uci.ics.texera.workflow.common.WorkflowContext
+import edu.uci.ics.texera.workflow.common.storage.OpResultStorage
+import edu.uci.ics.texera.workflow.common.workflow.PhysicalPlan
+
+import scala.collection.mutable
+import scala.jdk.CollectionConverters.CollectionHasAsScala
+
+case class WorkflowScheduler(workflowContext: WorkflowContext) {
+  var physicalPlan: PhysicalPlan = _
+  var regionPlan: RegionPlan = _
+  private var regionExecutionOrder: Iterator[Set[Region]] = _
+  private var opResultStorage: OpResultStorage = _
+  def addPhysicalPlan(physicalPlan: PhysicalPlan): Unit = {
+    this.physicalPlan = physicalPlan
+  }
+
+  def setStorage(opResultStorage: OpResultStorage): Unit = {
+    this.opResultStorage = opResultStorage
+  }
+
+  /**
+    * Update the total order of the regions to be executed, based on the current physicalPlan.
+    */
+  def updateSchedule(): Unit = {
```

Comment:
If this function is only called after `addPhyscialPlan`, can we just make it private and call it there?

Reply:
combined with updateSchedule